test/jdk/com/sun/management/DiagnosticCommandMBean/DcmdMBeanTest.java

```diff
@@ -100,14 +109,38 @@ static void printOperation(MBeanOperationInfo info) {
                     Descriptor desc3 =
                             (Descriptor)desc2.getFieldValue(desc2.getFieldNames()[j]);
                     for(int k=0; k<desc3.getFieldNames().length; k++) {
-                        System.out.println("\t\t\t"+desc3.getFieldNames()[k]+"="
-                                           +desc3.getFieldValue(desc3.getFieldNames()[k]));
+                        String fieldName3 = desc3.getFieldNames()[k];
+                        Object fieldValue3 = desc3.getFieldValue(fieldName3);
+                        System.out.print("\t\t\t" + fieldName3 + "=" + fieldValue3);
+                        if (fieldName3.equals("dcmd.arg.type")) {
+                            if (!isPublicType((String) fieldValue3)) {
+                                System.out.print("\t** FAILED ** not a standard type");
+                                failures++;
+                            }
+                        }
+                        System.out.println();
                     }
                 }
             } else {
                 System.out.println("\t"+desc.getFieldNames()[i]+"="
                         +desc.getFieldValue(desc.getFieldNames()[i]));
             }
         }
+        return failures;
+    }
+
+    // Knowledge of the types made public by com.sun.management.internal.DiagnosticCommandImpl
+    private static final String [] publicTypes = new String [] { "INT", "STRING", "BOOLEAN", "STRING SET", "MEMORY SIZE", "NANOTIME" };
```

Comment:
These type are "implementation dependent", yet we are referring to them as public types. Also, these "public types" are now in two different places. If they were public, only one copy should be needed. Maybe we just need better terminology since they are not actually public, but just happen to be the types the implementation is know to return...err, I guess you could say the leak out to the "public". Sigh. IDK. I guess I just feel we could do a better job in how we refer to these types in the code.